[Au Chen Xi, Gabriel] iP

Gabau opened the pull request:
# Sariel

>Be Free

Let's you do you. It's

 * text-based
 * easy to learn
 * ~~FAST~~ _SUPER_ _FAST_ to use

All you have to do is

 1. Download the jar from [here](https://github.com/Gabau/ip/releases)
 1. Java it in command prompt
 1. Add your tasks
 1. __PROFIT__

And its __FREE__ __OF__ __CHARGE__ 😂 

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines  (incomplete)
 - [ ] Reminders (incomplete)
 - [ ] Teleportation to Mars (incomplete)

----

If you are a Java prorammer, you can use it to practice java too.  
Heres the main method:   

```java
public class Main {
   public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Gabau committed:
Completed level 1 tasks.

Gabau committed:
Level 2. Add, List.

Gabau committed:
Level-3, added the task class and the ability to mark tasks as done with the done command.

Gabau committed:
Level 4, added Task class files and different task features.

Gabau committed:
Modified test files

Gabau committed:
Modified line when adding task

Gabau committed:
Level 5, handling errors and duplicate input.

Gabau committed:
Added ArrayList and removed NumberFormatException

Gabau committed:
Added delete and reorganised the run function

Gabau committed:
More tests

Gabau committed:
Reorganize Task classes.

It was too messy to deal with all the java files on the top layer.

Shifted the Task files into a packag util.tasks for convenience.

Gabau committed:
Setup encode and decode.

There was no capability for encoding and decoding the Tasks into a text document.

By adding such functions on the Task objects, it is easier to encode and decode the Tasks to save them.

Gabau committed:
Complete write and read methods for storing task information.

Did not have capability to read and write to a file to store the past tasks.

Added such functionality in this commit.

Gabau committed:
Add Local dates

Did not have any restrictions on the dates that could have been added.

This restrictions allows the dates to be modified in a more systematic manner.

Gabau committed:
Merge branch 'branch-Level-7'

Gabau committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

Gabau committed:
Add more OOP and Packages

There was a lack of OOP for the organisation of the program.

Finished creating the classes for more oop but not the packages.

Gabau committed:
Complete packages and more OOP

Gabau committed:
Fix Dlist

Gabau committed:
Add unit tests

Did not have any unit tests previously.

Included tests for parser, delcommand and DateTaskTable

Gabau committed:
Remove duplicate greeting

Gabau committed:
Add more javadocs

Gabau committed:
Improve javadoc

Gabau committed:
Improve coding standard

Gabau committed:
Add Find feature

Gabau committed:
Merge branch 'branch-A-CodingStandard'

Gabau committed:
Merge branch 'branch-A-JavaDoc'

Gabau committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/util/parser/Parser.java

Gabau committed:
Explicit imports

Gabau committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Gabau committed:
Obtain junit gradle support and compilation capability

Gabau committed:
Ammend checkstyle issues

Gabau committed:
Add JavaFx dependencies

Gabau committed:
Add basic GUI functionality

Gabau committed:
Add text output and task creation

Gabau committed:
Add listView viewer

Gabau committed:
Modify names of ui related methods and other variables

Gabau committed:
Add window resize features

Gabau committed:
Change the duke chat box position

Gabau committed:
Fix writing issue

Gabau committed:
Add some documentation for javafx parts

Gabau committed:
Ammend checkstyle issues

Gabau committed:
Merge tag 'branch-Level-10'

Gabau committed:
Merge branch 'branch-A-CheckStyle'

e0543978 commented on src/main/java/util/commands/AddCommand.java:
```diff
@@ -0,0 +1,40 @@
+import util.tasks.Task;
+import util.tasks.TaskList;
+import util.ui.Ui;
+/**
+ * Command representing the addition of a
+ * task to the task list.
+ *
+ */
```
You could consider removing the empty lines before and after your Javadocs since it's not necessary.

e0543978 commented on src/main/java/util/tasks/Task.java:
```diff
@@ -0,0 +1,145 @@
+    /**
+     * Marks when the task is done.
+     */
+    public void done() {
+        this.isDone = true;
+    }
```
Perhaps you could use a verb for your method name such as "setDone".

e0543978 commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,101 @@
+
+	----------------------------
+	----------------------------
+	Godspeed young padawan!
```
Love this

Gabau replied:
Thanks!

e0543978 reviewed (commented):
Code was tidy and easy to read aside from the minor code quality issues. All the best!

Gabau committed:
Change package directory

Gabau committed:
Fix checkstyle issues

Gabau committed:
Add error message to TaskInputWindow

There would be an exception thrown on the console before when the deadline or
event date was not selected.

Through the use of a null check, we are able to deal with the situation when
the date is not selected. It is made such that an error message will print,
but the content is not cleared.

It is done this way to prevent the case where some description has been
added and the error erases said description.

Let's
 * Error print out when date not selected
 * Clears content on submission.
 * Submission no longer closes task input window

Gabau committed:
Add assertions to methods

There was no explicit checks for many of the assumptions tha the methods made
about the parameters.

Add assertions to Task class files and to Task container files.

By adding the explicit checks, we reduce the chance of a bug slipping by undetected
as we reduce the number of incorrect assumptions we have of the code. Tasks and
their containers are selected as they will be the more frequent use case.

Let's
 * Add assertions to Task class
 * Add assertions to factory methods for Task and its subclasses

Gabau committed:
Improve CodeQuality

There were long method calls that were hard to follow.

Extracted methods from many of the long method calls.

By attempting to follow the Single level of abstraction principle, I have improved the
readability of the code.

Lets'
 * Extracted methods from long methods.

Gabau committed:
Merge pull request #2 from Gabau/branch-A-Assertions

Branch a assertions

Gabau committed:
Merge branch 'master' into branch-A-CodeQuality

Gabau committed:
Merge pull request #3 from Gabau/branch-A-CodeQuality

Improve CodeQuality

Gabau committed:
Improve search flexibility

Gabau committed:
Add reminders

Gabau committed:
Add continuous integration

Gabau committed:
Change logo

Gabau committed:
FIx duplication issue

Gabau committed:
Finish readme

Gabau committed:
Fix checkstyle

Gabau committed:
Switch to index.md

Gabau committed:
Remove text-ui tests

Gabau committed:
Fix url for resources

Gabau committed:
Modify readme and rename packages

Gabau committed:
Modify readme and rename packages

Gabau committed:
Fix checkstyle from incorrect ordering

Gabau committed:
Modify exception

Gabau committed:
Fix checkstyle

Gabau committed:
Fix width issue

Gabau committed:
Make resizable

Gabau committed:
Fix index

Gabau committed:
Remove logo printout

Gabau committed:
Change Ui.png

Gabau committed:
Add config.yml to docs

Gabau committed:
Set theme jekyll-theme-minimal

Gabau committed:
Remove yml

Gabau committed:
Merge remote-tracking branch 'origin/master'

Gabau committed:
Set index to link to user guide

Gabau committed:
Make docs easier to navigate

Gabau committed:
Change theme layout

Gabau committed:
Add marking tasks as done

Gabau committed:
Rename UseGuide.md

Gabau committed:
Switch from index to README

Gabau committed:
Add user guide link

Gabau committed:
Add exit by CLI function

Gabau committed:
More javadocs
